material_studies/rally/lib/finance.dart

```diff
@@ -0,0 +1,311 @@
+      amount: model.primaryAmount,
+    );
+  }
+
+  static FinancialEntityCategoryView fromBudgetItem(
+      BudgetData item, int i, BuildContext context) {
+    return FinancialEntityCategoryView(
+      suffix: Text(' LEFT',
+          style: Theme.of(context)
+              .textTheme
+              .body1
+              .copyWith(color: RallyColors.gray60, fontSize: 10.0)),
+      title: item.name,
+      subtitle: Formatters.usdWithSign.format(item.amountUsed) +
+          ' / ' +
+          Formatters.usdWithSign.format(item.primaryAmount),
+      indicatorColor: RallyColors.budgetColor(i),
+      indicatorFraction: item.amountUsed / item.primaryAmount,
+      amount: item.primaryAmount - item.amountUsed,
+    );
+  }
+
+  static List<FinancialEntityCategoryView> fromAccountDataList(
+      List<AccountData> items) {
+    return List<FinancialEntityCategoryView>.generate(items.length,
+        (i) => FinancialEntityCategoryViews.fromAccountItem(items[i], i));
+  }
+
+  static List<FinancialEntityCategoryView> fromBillDataList(
+      List<BillData> items) {
+    return List<FinancialEntityCategoryView>.generate(items.length,
+        (i) => FinancialEntityCategoryViews.fromBillItem(items[i], i));
+  }
+
+  static List<FinancialEntityCategoryView> fromBudgetItems(
+      List<BudgetData> items, BuildContext context) {
+    return List<FinancialEntityCategoryView>.generate(
+        items.length,
+        (i) =>
+            FinancialEntityCategoryViews.fromBudgetItem(items[i], i, context));
+  }
+}
+
+class FinancialEntityCategoryDetailsPage extends StatelessWidget {
+  final List<DetailedEventData> items =
+      DummyDataService.getDetailedEventItems();
+
+  @override
+  Widget build(BuildContext context) {
+    final List<_DetailedEventCard> cards = items
+        .map((i) => _DetailedEventCard(
+              title: i.title,
+              subtitle: Formatters.date.format(i.date),
+              amount: i.amount,
+            ))
+        .toList();
+
+    return Scaffold(
+      appBar: AppBar(
+        elevation: 0.0,
+        centerTitle: true,
+        title: Text(
+          'Checking',
+          style: Theme.of(context).textTheme.body1.copyWith(fontSize: 18.0),
+        ),
+      ),
+      body: Column(children: <Widget>[
+        SizedBox(
+            height: 200.0,
+            width: double.infinity,
+            child: RallyLineChart(events: items)),
+        Flexible(
+          child: ListView(shrinkWrap: true, children: cards),
+        )
+      ]),
+    );
+  }
+}
+
+class _DetailedEventCard extends StatelessWidget {
+  const _DetailedEventCard({
+    @required this.title,
+    @required this.subtitle,
+    @required this.amount,
+  });
+
+  final String title;
+  final String subtitle;
+  final double amount;
+
+  @override
+  Widget build(BuildContext context) {
+    return FlatButton(
+      onPressed: () {},
+      child: SizedBox(
+          height: 68.0,
+          child: Column(
+            children: <Widget>[
```

Comment:
Here again, removing `<Widget>` and adding commas will improve readability.

Reply:
Done